core/string.c

```diff
@@ -172,40 +172,21 @@ d_r_memmove(void *dst, const void *src, size_t n)
  */
 void *
 __memmove_chk(void *dst, const void *src, size_t n, size_t dst_len)
-#    ifdef MACOS
-/* OSX 10.7 gcc 4.2.1 doesn't support the alias attribute.
- * XXX: better to test for support at config time: for now assuming none on Mac.
- */
 {
     return memmove(dst, src, n);
 }
-#    else
-    __attribute__((alias("d_r_memmove")));
-#    endif
+
 void *
 __strncpy_chk(char *dst, const char *src, size_t n, size_t dst_len)
-#    ifdef MACOS
-/* OSX 10.7 gcc 4.2.1 doesn't support the alias attribute.
- * XXX: better to test for support at config time: for now assuming none on Mac.
- */
 {
     return strncpy(dst, src, n);
```

Comment:
While this works today due to the #define for Windows compat which does s/strncpy/d_r_strncpy/, I would prefer an explicit use of what function this is calling for readability.  This is ifdef UNIX already so no need for the Windows name.  So let's make this call d_r_strncpy.